24.md

```diff
@@ -0,0 +1,100 @@
+NIP-24
+======
+
+Private, Encrypted Direct Messages
+----------------------------------
+
+`draft` `optional` `author:jeffthibault`
+
+This NIP defines a scheme for clients to send and receive [NIP-04](https://github.com/nostr-protocol/nips/blob/master/04.md) direct messages more privately.
+
+Part 1: Decoy Key Proof Event
+-----------------------------
+
+A special event with kind 12, meaning `Decoy Key Proof` is defined similar to a NIP-04 event except the the unencrypted content MUST be a JSON-stringified object with the following structure:
+```
+{
+  "msg": "dk:<a 32 byte hex-encoded public key>",
+  "pk": "<32 byte hex-encoded public key of the event creator>",
+  "sig": "<64 byte signature of the sha256 hash of <msg>>"
+}
+```
+
+The event MUST contain only one `"p"` tag for the intended recipient.
+
+This event allows users to prove to someone else that they will send NIP-04 messages from a decoy public key and also verify when someone else attests to using a decoy public key for NIP-04 messages.  
+
+Part 2: Protocol
+----------------
+
+**Definitions**<br>
+`S[-]` = Sender's Nostr private key<br>
+`S[+]` = Sender's Nostr public key<br>
+`R[-]` = Recipient's Nostr private key<br>
+`R[+]` = Recipient's Nostr public key<br>
+`SR[ss]` = Sender and Recipient's shared secret = `ECDH(S[-], R[+])` or `ECDH(R[-], S[+])`<br>
+
+`S[d-]` = Sender's decoy private key<br>
+`S[d+]` = Sender's decoy public key<br>
+`R[d+]` = Recipient's decoy public key<br>
+
+`RDIH` is the `Recipients's Decoy Inbox Hash`. It is known only by the sender and recipient because it is defined as the 32 byte, hex-encoded digest of `SHA256(SHA256(<32 byte hex-encoded SR[ss]> + <32 byte hex-encoded R[+]>))`.
+
+### Sending
+
+**Step 1**:<br>
+For each new recipient, R, they message with, the sender MUST create a decoy private key, `S[d-]`, defined as `S[-] + SR[ss]`.
```

Comment:
`S[-]` and `SR[ss]` are two random integers, so the new secret, `S[d-]`, is their sum.

Reply:
@jb55 I'm not a cryptography person so I might be misunderstanding but I was thinking this would just use `secp256k1_ec_privkey_tweak_add` where `S[-]` is the private key and `SR[ss]` is the scalar. Any thoughts on this?

Reply:
@RandyMcMillan may be able to help